lib/SPIRVProducerPass.cpp

```diff
@@ -1927,6 +1927,11 @@ SPIRVID SPIRVProducerPassImpl::getSPIRVType(Type *Ty, bool needs_layout) {
     uint32_t bit_width =
         static_cast<uint32_t>(Canonical->getPrimitiveSizeInBits());
 
+    if (bit_width > 1) {
```

Comment:
I think this might need to be moved to `CanonicalType()`. As is, clspv might generate duplicate integer types which is invalid in SPIR-V.  For example if the kernel generated the i2 and also used i8, clspv would cache two different `OpTypeInt 8 0` instructions. Make sure to add a test that covers that situation.